### ISSUE TITLE
Dependency issue with netifaces 0.10.4

### ISSUE THREAD

**Issue by postlund** [jstasiak/python-zeroconf#103]
I am using python-zeroconf in my library, https://github.com/postlund/pyatv, but with release 0.19.0 I'm having issues resolving dependencies. It appears to be a well-known issue using this version with python3:

https://bitbucket.org/al45tair/netifaces/issues/13/0104-install-is-broken-on-python-3x
https://github.com/GNS3/gns3-server/issues/97

Since Home Assistant (where I use this library) has pinned version 0.19.0, I cannot just blacklist that version until a more permanent fix is in place (that's not desirable anyways). There's a special version of netifaces that claims to solve this issue (gns3-netifaces), but using the latest version of netifaces would of course be more desirable. Can any developer weigh in on this as it blocks a release that I want to make?

For reference, this is the backtrace I see:

```
Installed /home/postlund/pyatv/lib/python3.4/site-packages/six-1.10.0-py3.4.egg
Searching for netifaces<=0.10.4
Reading https://pypi.python.org/simple/netifaces/
Best match: netifaces 0.10.4
Downloading https://pypi.python.org/packages/18/fa/dd13d4910aea339c0bb87d2b3838d8fd923c11869b1f6e741dbd0ff3bc00/netifaces-0.10.4.tar.gz#md5=36da76e2cfadd24cc7510c2c0012eb1e
Processing netifaces-0.10.4.tar.gz
Writing /tmp/easy_install-87y982o4/netifaces-0.10.4/setup.cfg
Running netifaces-0.10.4/setup.py -q bdist_egg --dist-dir /tmp/easy_install-87y982o4/netifaces-0.10.4/egg-dist-tmp-1nh5a95c
Traceback (most recent call last):
  File "setup.py", line 42, in <module>
    'Topic :: Home Automation',
  File "/usr/lib/python3.4/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.4/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.4/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/home/postlund/pyatv/lib/python3.4/site-packages/setuptools/command/develop.py", line 32, in run
    self.install_for_development()
  File "/home/postlund/pyatv/lib/python3.4/site-packages/setuptools/command/develop.py", line 132, in install_for_development
    self.process_distribution(None, self.dist, not self.no_deps)
  File "/home/postlund/pyatv/lib/python3.4/site-packages/setuptools/command/easy_install.py", line 716, in process_distribution
    [requirement], self.local_index, self.easy_install
  File "/home/postlund/pyatv/lib/python3.4/site-packages/pkg_resources.py", line 631, in resolve
    dist = best[req.key] = env.best_match(req, ws, installer)
  File "/home/postlund/pyatv/lib/python3.4/site-packages/pkg_resources.py", line 874, in best_match
    return self.obtain(req, installer)
  File "/home/postlund/pyatv/lib/python3.4/site-packages/pkg_resources.py", line 886, in obtain
    return installer(requirement)
  File "/home/postlund/pyatv/lib/python3.4/site-packages/setuptools/command/easy_install.py", line 636, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "/home/postlund/pyatv/lib/python3.4/site-packages/setuptools/command/easy_install.py", line 666, in install_item
    dists = self.install_eggs(spec, download, tmpdir)
  File "/home/postlund/pyatv/lib/python3.4/site-packages/setuptools/command/easy_install.py", line 856, in install_eggs
    return self.build_and_install(setup_script, setup_base)
  File "/home/postlund/pyatv/lib/python3.4/site-packages/setuptools/command/easy_install.py", line 1078, in build_and_install
    self.run_setup(setup_script, setup_base, args)
  File "/home/postlund/pyatv/lib/python3.4/site-packages/setuptools/command/easy_install.py", line 1063, in run_setup
    run_setup(setup_script, args)
  File "/home/postlund/pyatv/lib/python3.4/site-packages/setuptools/sandbox.py", line 68, in run_setup
    DirectorySandbox(setup_dir).run(runner)
  File "/home/postlund/pyatv/lib/python3.4/site-packages/setuptools/sandbox.py", line 114, in run
    return func()
  File "/home/postlund/pyatv/lib/python3.4/site-packages/setuptools/sandbox.py", line 67, in runner
    _execfile(setup_script, ns)
  File "/home/postlund/pyatv/lib/python3.4/site-packages/setuptools/sandbox.py", line 43, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-87y982o4/netifaces-0.10.4/setup.py", line 17, in <module>
    description='Library for controlling an Apple TV',
AttributeError: 'dict' object has no attribute 'print'
```
(it's the good ol' module-mix-up problem)

**Comment by jstasiak** [comment]
The dependency will remain pinned (see https://github.com/jstasiak/python-zeroconf/issues/84 for details) until https://bitbucket.org/al45tair/netifaces/issues/39/netmask-is-always-255255255255 is resolved.

I believe you need an upgraded pip for the error mentioned above to be resolved.

**Comment by postlund** [comment]
Yeah, I noticed that a lot of people have had problems with 0.10.5 (which resolves my particular issue, but introduces other problems). I am already using the latest version of pip and that doesn't work either.

```
$ pip search pip
...
pip (9.0.1)                                          - The PyPA recommended tool for installing Python packages.
  INSTALLED: 9.0.1 (latest)
...
```
So I am a bit stuck here I'm afraid.

**Comment by MaralAfris** [comment]
Hi,

It seems that the netifaces guys are not so active, maybe moving to gn3-netifaces instead would be a solution ?
[https://pypi.python.org/pypi/gns3-netifaces/0.10.4.1]()
Here is an issue post on that: [https://github.com/GNS3/gns3-server/issues/97]()

**Comment by postlund** [comment]
@jstasiak I wrote ticket about this issue over at the netifaces project and Alastair (the author) was pretty fast at fixing the mentioned issue. So, netifaces 0.10.6 is out which should make it possible to lift the pinning to 0.10.4. Is it possible to get a new release done?

**Comment by jstasiak** [comment]
Yup, I just released zeroconf 0.19.1 which allows installing netifaces >= 0.10.6, I trust that helps.

**Comment by postlund** [comment]
Just tried it and it works flawlessly :ok_hand: I am so happy! Thank for you very much for the release!
